### PR TITLE
Allow Hive and other connectors use system roles

### PR DESCRIPTION
**Pull request by kokosing**
Allow Hive and other connectors use system roles

In order for catalog to use system roles, a connector is required to
**not** to provide connector access control. Otherwise, catalog roles will be
used.

To allow usage of system roles for hive and other connectors, allow-all security
can not bind any access control.

### CODE REVIEW

**Comment by kokosing** [comment]
This is still in progress, but early feedback would be appreciated.

**Review by ksobolew — COMMENTED** [review]
Looks good; not sure I understand the Guice changes entirely yet :)

Is it possible to have no connector access control and no global access control?

**Comment by kokosing** [comment]
I removed the guice code.

**Comment by findepi** [comment]
> To allow usage of system roles for hive and other connectors

How is the Hive connector going to use system roles?
How does this interact with `hive.security=sql-standard`?

**Comment by dain** [comment]
> > To allow usage of system roles for hive and other connectors
> 
> How is the Hive connector going to use system roles? How does this interact with `hive.security=sql-standard`?

A connector can either use system roles or connector role, but not both. We came to this design after building the mixed model, and it was super confusing for the user (and the code was a mess).  No one could think of a case where someone would want this, so we settled on this simplified model.  Additionally, we further simplified the model down to "if a connector has its own security model, then it uses connector roles, otherwise everything is at the system level".  This was for the same complexity reasons, and no real use cases for mixed support.

With these choices, the system is really simple, if a connector provides a connector access control then it gets all role calls, otherwise we send role methods to a system resource.  The one thing we all missed was the allow all connector access control which is used by many connectors to say "I'm not doing any security checks".  When this is installed it unfortunately blocks system roles, and this PR fixes that by not returning an access control at all when allow-all is selected.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/security/HiveSecurityModule.java [hunk 1]
```diff
@@ -25,26 +25,32 @@
 public class HiveSecurityModule
         extends AbstractConfigurationAwareModule
 {
+    public static final String LEGACY = "legacy";
```
Comment (Praveen2112) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/security/HiveSecurityModule.java:29]: nit: We can replace it with enum ? Like we do in the next commit.

**Review by Praveen2112 — APPROVED** [review]

**Comment by findepi** [comment]
> So to answer the root question. This does not interact with `hive.security=sql-standard`. Hive SQL standard security continues to use connector roles, but when you select allow-all you _can_ use a system access control with system roles.

Thanks for the explanation @dain.

And what if i do _not_ select allow-all (e.g. select legacy, or sql-standard) -- can i still use system access control with system roles?

**Comment by byungnam** [comment]
Does this enables GRANT REVOKE statements to all connectors or specific (hive, atop, raptor, iceberg) connectors only?

**Pull request merged**